app/src/main/java/com/hedvig/app/feature/home/ui/claimstatus/data/PillData.kt

```diff
@@ -0,0 +1,76 @@
+
+data class PillData(val text: String, val type: PillType) {
+
+    sealed class PillType {
+        data class Contained(val color: Color) : PillType()
+        object Outlined : PillType()
+    }
+
+    companion object {
+        @Composable
+        fun fromClaimStatus(claimStatus: HomeQuery.ClaimStatus): List<PillData> {
+            return when (claimStatus.status) {
+                ClaimStatus.SUBMITTED -> listOf(PillData("CLAIM", PillType.Outlined))
+                ClaimStatus.BEING_HANDLED -> listOf(PillData("CLAIM", PillType.Outlined))
+                ClaimStatus.CLOSED -> {
+                    when (claimStatus.outcome) {
+                        ClaimOutcome.PAID -> {
+                            mutableListOf<PillData>().apply {
+                                add(
+                                    PillData(
+                                        stringResource(R.string.claim_decision_paid),
+                                        PillType.Contained(MaterialTheme.colors.primary)
+                                    )
+                                )
+                                val payout = claimStatus.payout
+                                if (payout != null) {
+                                    add(
+                                        PillData(
+                                            "${payout.amount} ${payout.currency}",
+                                            PillType.Contained(ClaimStatusColors.Pill.paid)
+                                        )
+                                    )
+                                }
+                            }.toList()
+                        }
+                        ClaimOutcome.NOT_COMPENSATED -> listOf(
+                            PillData(
+                                stringResource(R.string.claim_decision_not_compensated),
+                                PillType.Contained(MaterialTheme.colors.primary)
+                            )
+                        )
+                        ClaimOutcome.NOT_COVERED -> listOf(
+                            PillData(
+                                stringResource(R.string.claim_decision_not_covered),
+                                PillType.Contained(MaterialTheme.colors.primary)
+                            )
+                        )
+                        else -> throw IllegalArgumentException()
+                    }
+                }
+                ClaimStatus.REOPENED -> listOf(
+                    PillData(
+                        stringResource(R.string.home_claim_card_pill_reopened),
+                        PillType.Contained(ClaimStatusColors.Pill.reopened)
+                    ),
+                    PillData(
+                        stringResource(R.string.home_claim_card_pill_claim),
+                        PillType.Outlined
+                    ),
+                )
+                ClaimStatus.UNKNOWN__ -> throw IllegalArgumentException()
```

Comment:
Are the claimStatus options (or any enum) supposed to be a non-breaking change in the GraphQL schema? Should we just not render anything instead of crashing when a new one is added? Probably yes right?

Reply:
What to do when encountering an unknown `enum`-case needs to be evaluated for each use-case - here, I'm not quite sure what's best. Check with Hiro, perhaps 🙂